lib/pass.ml

```diff
@@ -0,0 +1,75 @@
+open Core
+
+type fmt =
+  | Concrete (* valid concrete syntax (P4 or C) *)
+  | Sexps    (* s-expression format *)
+  | Coq      (* .v file *)
+  | Ocaml    (* .ml file *)
+
+type output =
+  { out_file: Filename.t;
+    out_fmt: fmt }
+
+type 'a cfg =
+  | Skip
+  | Run of 'a
+
+let is_skip : 'a cfg -> bool =
+  function
+  | Skip -> true
+  | _ -> false
```

Comment:
It seems like this could probably be dervied

Reply:
what do you mean?